### PR TITLE
🌱 Remove use of 'master' word from our codebase

### LINKED ISSUE #3281: Remove use of "master" word from our codebase
These are some of the uses that can be changed within this repository

- https://github.com/kubernetes-sigs/cluster-api/blob/2ac3728d26593f7c54520999477aad45934e1c59/test/framework/cluster_proxy.go#L229-L234
- https://github.com/kubernetes-sigs/cluster-api/blob/2ac3728d26593f7c54520999477aad45934e1c59/controllers/machinehealthcheck_targets.go#L40
- https://github.com/kubernetes-sigs/cluster-api/blob/2ac3728d26593f7c54520999477aad45934e1c59/docs/book/src/user/concepts.md "Joining master and worker nodes to the cluster"
- https://github.com/kubernetes-sigs/cluster-api/blob/2ac3728d26593f7c54520999477aad45934e1c59/controlplane/kubeadm/internal/workload_cluster.go#L49 (only the name of the variable, the label string change is tracked separately)
- https://github.com/kubernetes-sigs/cluster-api/blob/2ac3728d26593f7c54520999477aad45934e1c59/controlplane/kubeadm/main.go#L85
- https://github.com/kubernetes-sigs/cluster-api/blob/2ac3728d26593f7c54520999477aad45934e1c59/bootstrap/kubeadm/main.go#L87
- https://github.com/kubernetes-sigs/cluster-api/blob/2ac3728d26593f7c54520999477aad45934e1c59/main.go#L93
- https://github.com/kubernetes-sigs/cluster-api/blob/2ac3728d26593f7c54520999477aad45934e1c59/docs/book/src/user/quick-start.md `export MASTER_NODE_TYPE="t1.small"` should be changed to `CONTROLPLANE_NODE_TYPE` (cc @gianarb)

/kind cleanup
/wg naming
/milestone v0.3.x

### PR DESCRIPTION
**Pull request by SataQiu**
**What this PR does / why we need it**:
Remove use of `master` word from our codebase

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3281

### CODE REVIEW

File: bootstrap/kubeadm/main.go [hunk 1]
```diff
@@ -84,7 +84,7 @@ func InitFlags(fs *pflag.FlagSet) {
 		"Interval at which non-leader candidates will wait to force acquire leadership (duration string)")
 
 	fs.DurationVar(&leaderElectionRenewDeadline, "leader-election-renew-deadline", 10*time.Second,
-		"Duration that the acting master will retry refreshing leadership before giving up (duration string)")
+		"Duration that the acting control plane will retry refreshing leadership before giving up (duration string)")
```
Comment (vincepri) [inline bootstrap/kubeadm/main.go:87]:
```suggestion
		"Duration that the leading controller manager will retry refreshing leadership before giving up (duration string)")
```

File: controlplane/kubeadm/main.go [hunk 2]
```diff
@@ -82,7 +82,7 @@ func InitFlags(fs *pflag.FlagSet) {
 		"Interval at which non-leader candidates will wait to force acquire leadership (duration string)")
 
 	fs.DurationVar(&leaderElectionRenewDeadline, "leader-election-renew-deadline", 10*time.Second,
-		"Duration that the acting master will retry refreshing leadership before giving up (duration string)")
+		"Duration that the acting control plane will retry refreshing leadership before giving up (duration string)")
```
Comment (vincepri) [inline controlplane/kubeadm/main.go:85]:
```suggestion
		"Duration that the leading controller manager will retry refreshing leadership before giving up (duration string)")
```

File: docs/book/src/user/quick-start.md [hunk 3]
```diff
@@ -486,7 +486,7 @@ export NODE_OS="ubuntu_18_04"
 export SSH_KEY="my-ssh"
 export POD_CIDR="172.25.0.0/16"
 export SERVICE_CIDR="172.26.0.0/16"
-export MASTER_NODE_TYPE="t1.small"
+export CONTROLPLANE_NODE_TYPE="t1.small"
```
Comment (vincepri) [inline docs/book/src/user/quick-start.md:489]: CC @gianarb, we'll need to fix this environment variable in the packet provider

File: main.go [hunk 4]
```diff
@@ -90,7 +90,7 @@ func InitFlags(fs *pflag.FlagSet) {
 		"Interval at which non-leader candidates will wait to force acquire leadership (duration string)")
 
 	fs.DurationVar(&leaderElectionRenewDeadline, "leader-election-renew-deadline", 10*time.Second,
-		"Duration that the acting master will retry refreshing leadership before giving up (duration string)")
+		"Duration that the acting control plane will retry refreshing leadership before giving up (duration string)")
```
Comment (vincepri) [inline main.go:93]:
```suggestion
		"Duration that the leading controller manager will retry refreshing leadership before giving up (duration string)")
```

**Comment by vincepri** [comment]
Thanks for jumping on these changes @SataQiu !

File: docs/book/src/user/quick-start.md (hunk 3 above)
Comment (vincepri) [reply docs/book/src/user/quick-start.md:489]: Nevermind, saw the referenced commit above, seems we should be good to go here!

**Comment by fabriziopandini** [comment]
thanks @SataQiu !
/lgtm

**Pull request merged**